DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs

```diff
@@ -349,6 +349,41 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
                 capturedException.Throw();
             }
 
+            if (repositoryDevice != null && repositoryDevice.IsSimulatedDevice)
+            {
+                if (isEnabled)
+                {
+                    try
+                    {
+                        var securityKeys = await this.GetIoTHubKeysAsync(deviceId);
+                        await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
+                        {
+                            DeviceId = repositoryDevice.DeviceProperties.DeviceID,
+                            HostName = _configProvider.GetConfigurationSettingValue("iotHub.HostName"),
+                            Key = securityKeys.PrimaryKey
+                        });
+                    }
+                    catch (Exception ex)
+                    {
+                        //if we fail adding to table storage for the device simulator just continue
+                        Trace.TraceError("Failed to add enabled simulated device : {0}", ex.Message);
```

Comment:
maybe more specific logging? "Failed to add enabled device to simulated device storage. Device telemetry is expected not to be send." something like this ?